src/Datasource/EntityTrait.php

```diff
@@ -360,6 +360,20 @@ public function unsetProperty($property)
     }
 
     /**
+     * Get the list of all properties.
+     *
+     * The list of all properties is all standard properties
+     * plus virtual properties plus hidden properties.
+     *
+     * @return array A list of all entity properties
+     */
+    public function allProperties()
```

Comment:
I think if I didn't understand maybe i will not the only onee. Maybe another name ?